packages/api-server/api_server/routes/tasks/scheduled_tasks.py

```diff
@@ -159,18 +159,54 @@ async def del_scheduled_tasks_event(
     event_date: datetime,
     task_repo: TaskRepository = Depends(task_repo_dep),
 ):
+    logger.info(f"Deleting task with schedule id {task_id}, event date {event_date}")
+
     task = await get_scheduled_task(task_id)
     if task is None:
+        logger.error(f"Task with scehdule id {task_id} not found")
         raise HTTPException(404)
 
-    tz = ZoneInfo(app_config.timezone) if app_config.timezone else None
-    event_date_str = event_date.replace(tzinfo=tz).isoformat()
+    # Server time zone, if not defined to use UTC
+    server_tz_info = (
+        ZoneInfo(app_config.timezone) if app_config.timezone else ZoneInfo("UTC")
+    )
+    logger.info(f"Server tz: {server_tz_info}")
+
+    # If the event date has time zone information, we check if it is in the same
+    # time zone as the server
+    event_date_utc_offset = event_date.utcoffset()
+    server_time_utc_offset = datetime.now(tz=server_tz_info).utcoffset()
+    if event_date_utc_offset is not None and server_time_utc_offset is not None:
+        event_tz_utc_offset_seconds = round(event_date_utc_offset.total_seconds())
+        server_tz_utc_offset_seconds = round(server_time_utc_offset.total_seconds())
+        logger.info(
+            f"Event tz utc offset: {event_tz_utc_offset_seconds}, server tz utc offset: {server_tz_utc_offset_seconds}"
+        )
+
+        # if the time zones are the same, we can extract the date directly
+        if event_tz_utc_offset_seconds == server_tz_utc_offset_seconds:
+            logger.info("Event and server are in the same timezone")
+            event_date_str = event_date.isoformat()
+        # otherwise, we convert the date before extracting the date
+        else:
+            logger.info("Event and server are not in the same timezone")
```

Comment:
Don't think we need to log at info level here, debug level should be good.

Reply:
I'm avoiding switching logging level to debug for now, as that will flood the logs with all other information, for this particular fix I'll keep it to info first, but will revert it soon